### PR TITLE
Use psammead rich text transforms

### LINKED ISSUE #4087: Use rich-text-transforms utility for parsing candy_xml data
**Is your feature request related to a problem? Please describe.**
For the MAPs we need to find a way to render the rich text Ares provides us as `candy_xml`. Following https://github.com/bbc/psammead/issues/2301 we should use the package to handle rich text transforms.

**Describe the solution you'd like**
Using preprocessor rules convert the `candy_xml` into an Optimo block format and re-use the Blocks container to render the converted data. 

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Snapshot tests should cover the rendering of rich text. 

- ~[ ] This feature is expected to need manual testing.~

**Additional context**
A [POC PR](https://github.com/bbc/simorgh/pull/3989) does majority of this out of the box.

### PR DESCRIPTION
**Pull request by pjlee11**
~Resolves #4087~

**Overall change:** Adds the package `@bbc/psammead-rich-text-transforms` and uses it in a new preprocessor rule `convertToOptimoBlocks`.

**Code changes:**

- The new preprocessor rule takes the CPS data from Ares, looping over each of the blocks and has a callback function for each type in `typesToConvert`
- Handles paragraph conversion to prove the rich-text-transforms library is working as expected

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- ~[ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)~
- ~[] This PR requires manual testing~

### CODE REVIEW

**Comment by pjlee11** [comment]
image

Although the package is small when minified and gzipped it's install size is much larger than expected therefore we won't merge this into simorgh until we've picked a smaller parsing library in `psammead-rich-text-transforms`.

**Comment by pjlee11** [comment]
Blocked by https://github.com/bbc/psammead/issues/2363

File: scripts/bundleSizeConfig.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   // Size limit for all bundles used by each service (K)
   // Keep these +/- 5K and update frequently!
-  MIN_SIZE: 547,
-  MAX_SIZE: 585,
+  MIN_SIZE: 639,
+  MAX_SIZE: 680,
```
Comment (pjlee11) [inline scripts/bundleSizeConfig.js:5]: Increasing the bundle sizes by ~95KB is a significant increase. However, this is less than a single image on many of our pages and therefore I don't think it should be considered blocking. I've opened the issue https://github.com/bbc/simorgh/issues/4203 to address this in the medium-term.

Comment (dr3) [reply scripts/bundleSizeConfig.js:5]: I think we should investigate bundle splitting this prior to merging

**Review by dr3 — CHANGES_REQUESTED** [review]
I think we should investigate bundle splitting this prior to merging

File: src/app/models/propTypes/cpsAssetPage/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { bool, shape, string, number, any } from 'prop-types';
+// import { cpsMetadataPropTypes } from '../metadata';
```
Comment (ryanmccombe) [inline src/app/models/propTypes/cpsAssetPage/index.js]: can we delete the commented out code, or add a TODO comment if something needs done with it in the future

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+const handleMissingType = block =>
+  console.log(`Missing type field on block ${block.type}`); // eslint-disable-line no-console
+
+const typesToConvert = {
+  paragraph: convertParagraph,
+};
+
+const parseBlockByType = block => {
+  const { type } = block;
```
Comment (ryanmccombe) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks.js:15]: Small risk here with janky data - blocks array having an undefined entry will throw an exception

**Review by ryanmccombe — APPROVED** [review]
yay

File: scripts/bundleSizeConfig.js (hunk 1 above)
Comment (ryanmccombe) [reply scripts/bundleSizeConfig.js:5]: these bundle sizes are not really representative - shouldn't it be the production size we care about, after minification and gzip?

Comment (dr3) [reply scripts/bundleSizeConfig.js:5]: While its true that its not entirely representative of the gzipped size, its still a big indicator that this change is too big to pull into simorgh

**Review by ibMadbouly — APPROVED** [review]
LGr8TM 👍

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks.js (hunk 3 above)
Comment (pjlee11) [reply src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks.js:15]: Should now be covered by the `if` on line 12

File: scripts/bundleSizeConfig.js (hunk 1 above)
Comment (pjlee11) [reply scripts/bundleSizeConfig.js:5]: Attempting to bundle split this logic locally on the branch `attempt-bundle-splitting-for-CPS-asset`  which is [based off this branch](https://github.com/bbc/simorgh/compare/use-psammead-rich-text-transforms_signed-commits2...attempt-bundle-splitting-for-CPS-asset_signed-commits?expand=1)

**Comment by pjlee11** [comment]
Closing this as signed commits are now enforced

**Pull request closed without merging**